pkg/auth/kubesecret.go

```diff
@@ -0,0 +1,175 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+var kubeSecretListener *KubeSecretListener
+
+type KubeSecretListener struct {
+	dockerConfigs map[string]*configfile.ConfigFile
+	configMu      sync.Mutex
+	informer      cache.SharedIndexInformer
+}
+
+func InitKubeSecretListener(ctx context.Context, kubeconfigPath string) {
+	if kubeSecretListener == nil {
+		kubeSecretListener = NewKubeSecretListener(ctx, kubeconfigPath)
+	}
+}
+
+func NewKubeSecretListener(ctx context.Context, kubeconfigPath string) *KubeSecretListener {
```

Comment:
Is it necessary to export `NewKubeSecretListener`? I think `InitKubeSecretListener` and `NewKubeSecretListener` can be merged.